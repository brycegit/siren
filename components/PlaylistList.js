import React, { Component } from 'react';
import { StyleSheet, Text, View, TextInput, ScrollView, Image, Button} from 'react-native';
import { connect } from 'react-redux';
import { actionCreators as playerActions } from '../actions/Player';
import { actionCreators as swipeActions } from '../actions/Swipe';
import PlaylistCard from './PlaylistCard';
import PlaylistCardNoSwipe from './PlaylistCardNoSwipe';
import { actionCreators as playlistActions } from '../actions/Playlist';
import { getAllPlaylists } from '../helpers';

const mapStateToProps = (state) => ({
  allplaylists: state.playlist.allplaylists,
  token: state.main.token,
});

class PlaylistList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
      visible: false
    }
  }

  addPlaylist(){
    var that = this;
    fetch("http://siren-server.herokuapp.com/api/playlists/create-playlist", {
      method: "POST",
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
        'Authorization': this.props.token
      },
      body: JSON.stringify({name: this.state.text})
    })
    .then(function(response){
      that.setState({text: ""});
      if(response){
        getAllPlaylists(that.props);
      }
    });
  }

  componentDidMount(){
    if(!this.props.allplaylists.length){
      getAllPlaylists(this.props);
    }
  }

  render() {
   return (
      <View style={styles.mainView}>
        <ScrollView>
        <View style={styles.searchArea}>
          <Text>Create New Playlist</Text>
          <View style={styles.searchBar}>
            <TextInput underlineColorAndroid='rgba(0,0,0,0)' style={styles.searchInput} onChangeText={(text) => {this.setState({text});}} onSubmitEditing={() => getAllPlaylists(this.props)} value={this.state.text}/>
            <Button style={styles.button} onPress={this.addPlaylist.bind(this)} underlayColor='#99d9f4' title='Create' />
          </View>
        </View>
        {this.props.allplaylists.slice(0,2).map((playlist, index) => {
          return (
            <PlaylistCardNoSwipe key={index} playlist={playlist}/>
          )
        })}
          {this.props.allplaylists.slice(2).map((playlist, index) => {
            return (
              <PlaylistCard key={index + 1} playlist={playlist}/>
            )
          })}
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  mainView:{
    marginBottom: 185,
  },
  searchArea: {
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  episodeList:{
    width: '100%',
    marginBottom: 210,
  },
  searchInput: {
    height: 40,
    width: 200,
    borderColor: 'gray',
    borderWidth: 1,
    padding: 10
  },
  searchBar: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    margin: 10,
  },
  searchButton: {
    marginLeft: 8,
  },
})

export default connect(mapStateToProps)(PlaylistList);
