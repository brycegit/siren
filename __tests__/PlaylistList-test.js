import React from 'react';
import PlaylistList from '../components/PlaylistList.js';
import * as renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../reducers/index';
const store = createStore(reducer);

//TESTING REACT COMPONENT
test('renders correctly', () => {
  const tree = renderer.create(
    //wrap component in provide so it has access to store
    <Provider store={store}>
      <PlaylistList />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


//To retake snapshot: npm test -- -u PlaylistList
