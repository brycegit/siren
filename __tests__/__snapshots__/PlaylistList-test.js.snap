// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 185,
    }
  }
>
  <ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Create New Playlist
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 10,
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          style={
            Object {
              "borderColor": "gray",
              "borderWidth": 1,
              "height": 40,
              "padding": 10,
              "width": 200,
            }
          }
          underlineColorAndroid="rgba(0,0,0,0)"
          value=""
        />
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Create
            </Text>
          </View>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
